### DISCUSSION TITLE
elog 拉取语雀文章，开头的 Front-matter被删除，导致hexo渲染网页，文章没有标题

### DISCUSSION THREAD

**Discussion by zhongfeiorz** [LetTTGACO/elog#66] [Q&A] [answered]
## 博客架构
elog+语雀+hexo+butterfly
## 问题描述
elog 拉取语雀文章，开头的 Front-matter被删除，导致hexo渲染网页，文章没有标题，图片示例如下
![语雀文章内容](https://github.com/LetTTGACO/elog/assets/76213159/8fc4c7f9-86e7-4cb1-a343-6e92809b3caf)
![elog拉取之后](https://github.com/LetTTGACO/elog/assets/76213159/e3066bb1-848c-4595-af93-cf8cb7eff58d)
## 怀疑
怀疑是elog 在解析语雀文档时，自动将文章的 Front-matter 解析了，导致生成的 markdown 文件，没有 Front-matter 部分
查看 elog sdk-yuque 源码，发现文章 body 确实被替换了，代码图片如下
![article body is replaced](https://github.com/LetTTGACO/elog/assets/76213159/20d3beb2-683c-4562-8642-21d4c04b5a28)
## 期望结果
而我想要的是 将语雀文章的  Front-matter 内容保留下来，输出到 markdown 文档中，这样在使用 hexo+butterfly 解析的时候，能够读到 markdown 的 Front-matter  内容，进而在网页上可以正常显示标题等内容

由于没有接触过前端开发，想问一下：
1. 要实现这个结果，需要怎么修改呢
2. 或者 elog 是否有设置，可以保留语雀的 Front-matter 到 对应的 markdown 文件中呢

非常感谢！

**Comment by LetTTGACO** [comment, marked as answer]
有的，请看文档：https://elog.1874.cool/notion/raqyleng501h23p1

**Reply by zhongfeiorz** [reply]
好的，谢谢开发者大大
